### PR TITLE
Ryujit/ARM32: Implement Lowering::LowerCast for ARM

### PR DESCRIPTION
**Pull request by hqueue**
Related issue: #8519 

Simple integer to interger type conversion is passed.

Test with ` System.Diagnostics.Contracts.Contract:Assert(bool)
`
**Before:** NYI assertion when lowering GT_CAST.
```
...
lowering call (before):
N001 ( 16, 10) [000036] --C-G-?------                call help void   HELPER.CORINFO_HELP_DBG_IS_JUST_MY_CODE


objp:
======

args:
======

late:
======
lowering call (after):
N001 ( 16, 10) [000036] --C-G-?------                call help void   HELPER.CORINFO_HELP_DBG_IS_JUST_MY_CODE




System.Diagnostics.Contracts.Contract:Assert(bool) - NYI (/home/hk0110/work/dxl/dotnet/github/hqueue/coreclr/src/jit/lowerarm.cpp:38 - NYI_ARM: ARM Lowering for cast)
ISSUE: <ASSERT> /home/hk0110/work/dxl/dotnet/github/hqueue/coreclr/src/jit/lowerarm.cpp (38) - Assertion failed 'NYI_ARM: ARM Lowering for cast' in 'System.Diagnostics.Contracts.Contract:Assert(bool)' (IL size 20)
```

**After:** Simple integer conversion  is passed
```
lowering call (before):
N001 ( 16, 10) [000036] --C-G-?------                call help void   HELPER.CORINFO_HELP_DBG_IS_JUST_MY_CODE


objp:
======

args:
======

late:
======
lowering call (after): 
N001 ( 16, 10) [000036] --C-G-?------                call help void   HELPER.CORINFO_HELP_DBG_IS_JUST_MY_CODE


LowerCast for: N002 (  3,  3) [000041] ----------S--             *  cast      int <- bool <- int

lowering call (before):
N009 (  1,  1) [000020] -------------       t20 =    const     int    4

N010 (  1,  1) [000021] -------------       t21 =    const     ref    null

N011 (  1,  1) [000022] -------------       t22 =    const     ref    null

N012 (  1,  1) [000023] -------------       t23 =    const     ref    null

                                                 /--*  t20    int    arg0 in r0
                                                 +--*  t21    ref    arg1 in r1
                                                 +--*  t22    ref    arg2 in r2
                                                 +--*  t23    ref    arg3 in r3
N017 ( 20, 18) [000024] --C-G--------             *  call      void   hackishModuleName.hackishMethodName

(and continues...)
```

### CODE REVIEW

**Comment by hqueue** [comment]
@dotnet/jit-contrib PTAL. Thanks!

File: src/jit/lowerarm.cpp [hunk 1]
```diff
@@ -32,10 +32,78 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
 #include "lower.h"
 #include "lsra.h"
 
-/* Lowering of GT_CAST nodes */
+/* Lower GT_CAST(srcType, DstType) nodes.
```
Comment (pgavlin) [inline src/jit/lowerarm.cpp]: Please use C++-style comments (`//`)

**Comment by pgavlin** [comment]
LGTM aside from a nit regarding C++-style comments.

File: src/jit/lowerarm.cpp (hunk 1 above)
Comment (hqueue) [inline src/jit/lowerarm.cpp]: Thank you for the advice :)

**Comment by CarolEidt** [comment]
Thanks for modifying the comment! It should really conform to the coding conventions as described here: https://github.com/dotnet/coreclr/blob/master/Documentation/coding-guidelines/clr-jit-coding-conventions.md#9.4, though I'm OK with making that change later. But could you merge the commits?

**Comment by hqueue** [comment]
> though I'm OK with making that change later. But could you merge the commits?
@CarolEidt You are right and I will read it again :)  I've also try to pass format-checker in CI.
Is there something remaining to satisfy coding convention?  or do you mean squashing commits ?

**Comment by hqueue** [comment]
@CarolEidt I found what I've missed after reading convention again.  I've applied conventions. Thank you for the review :)

**Comment by CarolEidt** [comment]
@hqueue - yes, I meant "squashing commits". Sorry for the confusion.

**Comment by hqueue** [comment]
@CarolEidt No problem, I also want to make it clear :)

**Comment by hqueue** [comment]
@dotnet-bot test Linux ARM Emulator Cross Debug Build

**Comment by CarolEidt** [comment]
LGTM - thanks for updating comments and squashing commits

**Pull request merged**